lib/vrl/DESIGN.md

```diff
@@ -0,0 +1,423 @@
+# Vector Remap Language: Guiding Design Principles
+
+This document describes the high-level goals and directions of the Vector Remap
+Language (VRL). It is intended to help guide discussions and decisions made
+during the development of VRL itself. The document is less relevant to _users_
+of VRL, the [language documentation][docs] exists for that audience.
+
+[docs]: https://vrl.dev
+
+## Table of Contents
+
+<!-- vim-markdown-toc GFM -->
```

Comment:
Totally off topic, but what does this line do?

Reply:
They are markers for the Vim plugin I use to generate a table of contents. I usually keep it in while I develop the RFC/document so that the TOC gets auto-updated as I edit the rest of the document, and then remove them before I merge the document.

(I just removed them in the final commit, as I'm getting ready to merge)